source/usage-examples/bulkWrite.txt

```diff
@@ -0,0 +1,93 @@
+  </reference/write-concern>` and order.
+
+  By default, MongoDB executes bulk operations in the specified order
+  (i.e. serially). When executing operations in order, if an error occurs
+  during the processing MongoDB returns without processing any of the
+  remaining operations in the list. In contrast, when ``ordered`` is
+  ``false``, MongoDB continues to process remaining write operations in
+  the list. Unordered operations are faster as MongoDB can execute the
+  operations in parallel, but the results of the operation may vary. For
+  example, a ``deleteOne`` operation run before an ``updateMany``
+  operation might have a different result from running it after the
+  ``updateMany``. Refer to :ref:`Execution of Operations
+  <bulkwrite-write-operations-executionofoperations>` for more
+  information.
+
+- ``callback``: command result callback. Like other collection methods,
+  ``bulkWrite()`` returns a Promise if you do not specify a callback. The
+  Promise resolves to a :node-api:`bulkWriteOpCallback </Collection.html#~bulkWriteOpCallback>` object containing the
+  result of the operation.
+
+``bulkWrite()`` returns a ``BulkWriteResult`` object
+that includes the number of updated or inserted documents, the ``_id`` of
+each document, and any ``writeError`` or ``writeConcernErrors``. Refer to
+the documentation for :node-api:`BulkWriteResult </BulkWriteResult>` for
+more information.
+
+If you create an index with a :ref:`unique constraint
+<index-type-unique>`, you might encounter a duplicate key write error
+during an operation. The following example shows a duplicate key error
+encountered when two of the users in the inserted sample dataset had the
+same email address. 
+
+.. code-block:: sh
+
+   Error during bulkWrite, BulkWriteError: E11000 duplicate key error collection: sample_mflix.users index: email_1 dup key: { : "rita_91@example.com" }
+
+Similarly, if you attempt to perform a bulk write against a  collection
+that uses :manual:`schema validation </core/schema-validation>`, you may
+encounter warnings or errors related to the formatting of inserted or modified
+documents.
+
+Example
+-------
+
+The following code sample performs a bulk write operation against the
+``users`` collection in the ``sample_mflix`` database. Specifically, the
+program formats the data from a ``users.json`` source file to perform a
+series of ``insertOne`` operations with the ``bulkWrite()``collection
+method. Download the dataset here: `users.json
+<https://raw.githubusercontent.com/mongodb-university/universal-driver-examples/master/users.json>`_
+
+``users.json`` is an array of objects that resemble the following:
+
+.. code-block:: javascript
+
+   { 
+     "email" : "marieconrad@example.com",
+     "password" : "450f6704710dcf8033157978f7fbdfde",
+     "name" : "Marie Conrad"
+   }
+
+The existing documents in the ``users`` collection also contain
+``email``, ``password``, and ``name`` fields.
```

Comment:
This sentence seems kind of lonely on its own between two code samples. Maybe we could combine it with the sentence before the example document by just mentioning that the ``users.json`` documents have the same structure as the documents already in the ``users`` collection?

Reply:
Fair. Tweaked the intro to the sample doc.